### PR TITLE
[TEVA-1296] add UI to manage filters of a subscription

### PR DESCRIPTION
**Pull request by alexbowen**
https://dfedigital.atlassian.net/browse/TEVA-1296

Screenshot 2020-10-14 at 06 35 30
Screenshot 2020-10-14 at 06 35 46

### CODE REVIEW

File: app/controllers/subscriptions_controller.rb [hunk 1]
```diff
@@ -30,15 +30,18 @@ def create
...
     @subscription = SubscriptionPresenter.new(subscription)
+    @subscription_update_form = SubscriptionForm.new(subscription)
 
     Auditor::Audit.new(subscription, 'subscription.alert.edit', current_session_id).log_without_association
   end
 
   def update
     subscription = Subscription.find_and_verify_by_token(token)
     @subscription = SubscriptionPresenter.new(subscription)
+    @subscription_update_form = SubscriptionForm.new(subscription)
```
Comment (cpjmcquillan) [inline app/controllers/subscriptions_controller.rb:41]: At this point the form needs to take the params as arguments

File: app/controllers/subscriptions_controller.rb [hunk 2]
```diff
@@ -30,15 +30,18 @@ def create
...
 
-    if @subscription.update(subscription_params)
+    if @subscription_update_form.valid?
```
Comment (cpjmcquillan) [inline app/controllers/subscriptions_controller.rb:43]: Need to add validations on the form for 
- Email presence, email
- Frequency presence
- At least one other field not blank

File: app/controllers/subscriptions_controller.rb [hunk 3]
```diff
@@ -30,15 +30,18 @@ def create
...
+      @subscription.update(@subscription_update_form.to_hash)
```
Comment (cpjmcquillan) [inline app/controllers/subscriptions_controller.rb:44]: `@subscription.update` needs to be given a hash like this
```ruby
{ email: 'e@mail.com', frequency: 'weekly', search_criteria: { keyword: 'maths', etc etc } }
```

File: app/controllers/subscriptions_controller.rb [hunk 4]
```diff
@@ -30,15 +30,18 @@ def create
   def edit
     subscription = Subscription.find_and_verify_by_token(token)
     @subscription = SubscriptionPresenter.new(subscription)
```
Comment (cpjmcquillan) [inline app/controllers/subscriptions_controller.rb:32]: This and the line above can be in a `before_action` method

**Review by cesidio — CHANGES_REQUESTED** [review]
There seems to be a bug. When I make a location search for say `London`, and I try to make a new subscription, the keyword takes London too, even though it should just the be location

**Comment by cpjmcquillan** [comment]
> There seems to be a bug. When I make a location search for say `London`, and I try to make a new subscription, the keyword takes London too, even though it should just the be location

It's because the application doesn't have the LocationPolygons imported

File: app/controllers/subscriptions_controller.rb [hunk 5]
```diff
@@ -50,28 +50,25 @@ def update
...
-  def token
-    ParameterSanitiser.call(params).require(:id)
+  def origin_param
+    params.permit(:origin)
   end
 
-  def subscription_params
-    ParameterSanitiser.call(params.require(:subscription)).permit(:email, :frequency, :search_criteria)
+  def search_criteria_params
+    ParameterSanitiser.call(params.require(:search_criteria).permit(:keyword, :location, :radius, job_roles: [], phases: [], working_patterns: []))
```
Comment (alexbowen) [inline app/controllers/subscriptions_controller.rb:64]: am i right in thinking this was the part that was making those job_roles etc appear incorrectly on confirmation page?

**Comment by alexbowen** [comment]
@cpjmcquillan you are a superhero helping me get this over the line 🥇

**Review by cesidio — APPROVED** [review]

**Pull request merged**